include/deal.II/base/tensor_base.h

```diff
@@ -116,6 +116,14 @@ class Tensor<0,dim,Number>
   Tensor (const Tensor<0,dim,Number> &);
 
   /**
+   * Copy constructor from tensors with different underlying scalar
+   * type. This obviously requires that the @p OtherNumber type is
+   * convertible to @p Number.
+   */
+  template <typename OtherNumber>
+  Tensor (const Tensor<0,dim,OtherNumber> &);
+
```

Comment:
Any reason not to make this constructor explicit as for Tensor<rank,dim>? Do we need to allow the conversion float -> Tensor<0,dim,float> -> Tensor<0,dim,double> this way?